src/components/orator/js/Orator.js

```diff
@@ -0,0 +1,571 @@
+     *                           // enclosing DOM element
+     *      node: {DomNode}, // the current child node being parsed
+     *      childIndex: {Integer}, // the index of the child node being parsed relative to its parent
+     *      parentNode: {DomElement}, // the parent DOM node
+     *      word: {String} // the text, `word`, parsed from the node. (It may contain only whitespace.)
+     *   }
+     * @param {ParseQueue[]} parsed - An array of data points for the Parsed DOM
+     * @param {String} word - The word, parsed from the node, to be added
+     * @param {DomNode} childNode - The current textnode being operated on
+     * @param {Integer} blockIndex - The index into the entire block of text being parsed from the DOM
+     * @param {Integer} charIndex - The index into the current node being operated on, that is the start index of the
+     *                              word in the string representing the text of the node.
+     * @param {Integer} childIndex - The index of the node in the list of its parent's child nodes.
+     */
+    fluid.orator.domReader.addParsedData = function (parsed, word, childNode, blockIndex, charIndex, childIndex) {
+        parsed.push({
+            blockIndex: blockIndex,
+            startOffset: charIndex,
+            endOffset: charIndex + word.length,
+            node: childNode,
+            childIndex: childIndex,
+            parentNode: childNode.parentNode,
+            word: word
+        });
+    };
+
+    /**
+     * Recursively parses a DOM element and it's sub elements to construct an array of data points representing the
+     * words and space between the words. This data structure provides the means for locating text to highlight as the
+     * self voicing engine runs.
+     * NOTE: consecutive whitespace is collapsed to the first whitespace character.
+     * NOTE: hidden text is skipped.
+     *
+     * @param {jQuery|DomElement} elm - the DOM node to parse
+     * @param {Integer} blockIndex - The `blockIndex` represents the index into the entire block of text being parsed.
+     *                              It defaults to 0 and is primarily used internally for recursive calls.
+     *
+     * @return {ParseQueue[]} - An array of data points for the Parsed DOM
+     *                          See fluid.orator.domReader.addParsedData for details on the structure.
+     */
+    fluid.orator.domReader.parse = function (elm, blockIndex) {
+        var parsed = [];
+        elm = fluid.unwrap(elm);
+        blockIndex = blockIndex || 0;
+
+        if (fluid.orator.domReader.hasTextToRead(elm)) {
+            var childNodes = elm.childNodes;
+
+            $.each(childNodes, function (childIndex, childNode) {
+                if (childNode.nodeType === fluid.orator.domReader.nodeType.TEXT_NODE) {
+                    var words = childNode.textContent.split(/(\s+)/); // split on whitespace, and capture whitespace
```

Comment:
Stick this "split" action into an invoker so that it can be a configurable policy

Reply:
In speaking with @amb26 today it was suggested to make a domParser subcomponent on the domReader. We'd put in all the specific parsing functions into that subcomponent.